### ISSUE TITLE
cumsum(x,axis=-1) does not work on random array

### ISSUE THREAD

**Issue by udibr** [Theano/Theano#2715]
When working on an array generated by random number generator, it returns x unchanged. If you use a positive value for axis then it works correctly but if axis=-1 it does not apply itself to the last dimension
```python
import theano
import theano.tensor as T
from theano.sandbox.rng_mrg import MRG_RandomStreams
import pprint

rng = MRG_RandomStreams(3)
u = rng.uniform(size=(2,5))
c = T.cumsum(u,axis=-1)
d = T.cumsum(u,axis=1)
f = theano.function([], outputs=[u, c, d], allow_input_downcast=True)

pprint.pprint(f())
```
[array([[ 0.005768  ,  0.00573857,  0.79128766,  0.73698175,  0.58187205],
       [ 0.62578964,  0.73792672,  0.21239796,  0.24640237,  0.14918256]], dtype=float32),
 array([[ 0.005768  ,  0.00573857,  0.79128766,  0.73698175,  0.58187205],
       [ 0.62578964,  0.73792672,  0.21239796,  0.24640237,  0.14918256]], dtype=float32),
 array([[ 0.005768  ,  0.01150657,  0.80279422,  1.53977597,  2.12164807],
       [ 0.62578964,  1.36371636,  1.5761143 ,  1.82251668,  1.97169924]], dtype=float32)]

**Comment by nouiz** [comment]
@MarcCote
<https://github.com/Theano/Theano/issues?q=is%3Apr+is%3Aopen+author%3AMarcCote>
Can you look at it? I think it should raise an error or work as expected.

**Comment by MarcCote** [comment]
The problem is not related to the use of random arrays but if the GPU version is used or not.

While implementing the GPU version, I didn't realize the function `cumsum` accepted negative number for its parameter `axis` even though it entirely makes sense. I fixed this issue in PR #2716
